### PR TITLE
TW-1886 update chatlist UI part1

### PR DESCRIPTION
**Pull request by KhaledNjim**
## Ticket
#1886 

Figma design : [Figma](https://www.figma.com/design/uLw3iWnOEGaRlJnfPz4Pv0/Twake-[Matrix]?node-id=25783-37873&t=n6R8XpD6p1Tskyo3-0)

## Test recommendations
 Test on all platform with different screen sizes

## Resolved
**Attach screenshots or videos demonstrating the changes**
- Responsive Web:

https://github.com/user-attachments/assets/0c69a919-542d-4eb1-8e7c-0108e56c365d

image

- Android
image

-multi account and changing avatar demo:

[multiAccountDemo.webm](https://github.com/user-attachments/assets/50dfd56f-5c21-45a1-8dc1-ded73933976e)

DOD:
- [x] move avatar to bottom navigation bar
- [x] update bottom navigation bar icons
- [x] center title

### CODE REVIEW

**Comment by nqhhdev** [comment]
- Pls update the video demo for this case.

File: lib/widgets/layouts/enum/adaptive_destinations_enum.dart [hunk 1]
```diff
@@ -41,4 +68,30 @@ enum AdaptiveDestinationEnum {
...
   }
+
+  void initProfileNotifier(BuildContext context) async {
+    final profile =
+        await Matrix.of(context).client.fetchOwnProfile(getFromRooms: false);
+    profileNotifier.value = profile;
+  }
+
+  Widget getAvatar(BuildContext context, bool isSelected) {
+    initProfileNotifier(context);
```
Comment (nqhhdev) [inline lib/widgets/layouts/enum/adaptive_destinations_enum.dart:79]: IMO: This is a bad thing! Use the correct life cycle of a widget.

File: lib/widgets/layouts/enum/adaptive_destinations_enum.dart [hunk 2]
```diff
@@ -1,35 +1,62 @@
...
+          ),
           label: L10n.of(context)!.chats,
         );
       case AdaptiveDestinationEnum.settings:
         return NavigationDestination(
-          icon: const TwakeNavigationIcon(
-            icon: Icons.settings,
-          ),
+          icon: _responsive.isMobile(context)
+              ? getAvatar(context, false)
```
Comment (nqhhdev) [inline lib/widgets/layouts/enum/adaptive_destinations_enum.dart:50]: Should demo for multi account

File: lib/pages/chat_list/chat_list_header.dart [hunk 3]
```diff
@@ -30,12 +31,21 @@ class ChatListHeader extends StatelessWidget {
...
           padding: ChatListHeaderStyle.searchInputPadding,
           child: PlatformInfos.isWeb
               ? _normalModeWidgetWeb(context)
               : _normalModeWidgetsMobile(context),
         ),
+        if (ChatListHeaderStyle.responsive.isMobile(context))
+          Divider(
+            height: ChatListHeaderStyle.dividerHeight,
+            thickness: ChatListHeaderStyle.dividerThickness,
+            color: LinagoraSysColors.material().surfaceTint.withOpacity(0.16),
```
Comment (nqhhdev) [inline lib/pages/chat_list/chat_list_header.dart:47]: IMO: Should usse `LinagoraStateLayer().opacityLayer3`

File: lib/pages/chat_list/chat_list_header_style.dart [hunk 4]
```diff
@@ -9,13 +9,14 @@ class ChatListHeaderStyle {
   static ResponsiveUtils responsive = getIt.get<ResponsiveUtils>();
 
   static const double searchRadiusBorder = 24.0;
-  static const double searchBarContainerHeight = 64.0;
+  static const double searchBarContainerHeight = 57.0;
```
Comment (nqhhdev) [inline lib/pages/chat_list/chat_list_header_style.dart:12]: Why do you change it?

File: lib/widgets/layouts/enum/adaptive_destinations_enum.dart [hunk 5]
```diff
@@ -1,35 +1,62 @@
...
 import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:matrix/matrix.dart';
 
 enum AdaptiveDestinationEnum {
   contacts,
   rooms,
   settings;
 
+  static ValueNotifier<Profile> profileNotifier =
+      ValueNotifier(Profile(userId: ''));
```
Comment (sherlockvn) [inline lib/widgets/layouts/enum/adaptive_destinations_enum.dart:19]: why you need notifier here ?

**Comment by sherlockvn** [comment]
Please demo the case of responsive

**Comment by sherlockvn** [comment]
The counter is cut in iOS
image

File: lib/config/themes.dart [hunk 6]
```diff
@@ -267,7 +268,9 @@ abstract class TwakeThemes {
         iconTheme: IconThemeData(
           size: iconSize,
           color: brightness == Brightness.light
-              ? LinagoraSysColors.material().onBackground
+              ? isColumnMode(context)
```
Comment (sherlockvn) [inline lib/config/themes.dart:271]: be careful on changing in this file, because it may effect the whole color of the app

Comment (sherlockvn) [reply lib/config/themes.dart:271]: looks like it effects the color of bottom color 
IMG_A81DC35FF18F-1

File: lib/pages/chat_list/chat_list_header_style.dart (hunk 4 above)
Comment (KhaledNjim) [reply lib/pages/chat_list/chat_list_header_style.dart:12]: it has 57 height in  figma

File: lib/widgets/layouts/enum/adaptive_destinations_enum.dart (hunk 5 above)
Comment (KhaledNjim) [reply lib/widgets/layouts/enum/adaptive_destinations_enum.dart]: removed

File: lib/config/themes.dart (hunk 6 above)
Comment (KhaledNjim) [reply lib/config/themes.dart]: fixed

File: lib/widgets/avatar/bottom_navigation_avatar.dart [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+                width: isSelected ? 2 : 0,
+                color: LinagoraSysColors.material().primary,
+              ),
+            ),
+          ),
+          child: Avatar(
+            name: profile?.displayName,
+            mxContent: profile?.avatarUrl,
+            size: 24,
+            fontSize: 10,
```
Comment (nqhhdev) [inline lib/widgets/avatar/bottom_navigation_avatar.dart:33]: Create a style file for it

**Comment by nqhhdev** [comment]
I found problems on web responsive related to UI

Screenshot 2024-07-16 at 14 48 57

**Comment by sherlockvn** [comment]
i think this case should have the avatar in the icon, right ?
image

**Comment by Te-Z** [comment]
> I found problems on web responsive related to UI
> 
> Screenshot 2024-07-16 at 14 48 57 Screenshot 2024-07-16 at 14 51 13

Hi  @Bobpodvalnyi WDYT ? Should we remove the hover on tablet or not ?

**Comment by nqhhdev** [comment]
I found the bug: Related to counter icon

Screenshot 2024-07-17 at 19 01 05

**Comment by Bobpodvalnyi** [comment]
This is not a bug, it should be the case for two-digit numbers and more the counter should be stretched out

**Comment by Bobpodvalnyi** [comment]
@KhaledNjim what happened to the indentation between the icon and the text? I can see the difference even on the screenshot. Please make the indentation smaller

**Comment by Te-Z** [comment]
Hi @KhaledNjim could you update your demo videos please ?

**Comment by Te-Z** [comment]
Screenshot_20240724-055944

When switching from one account to the other the avatar doesn't change

**Comment by sherlockvn** [comment]
i tested it again in iOS, everything's good, but the splash ink around the icon when i tap, its overlapped, it should only have 1 ink.
https://github.com/user-attachments/assets/44717173-ac69-44c1-af03-1739960054cd

**Comment by sherlockvn** [comment]
When adding new account, the avatar in the bottom sheet not changed too

https://github.com/user-attachments/assets/81023ca1-84b7-4f67-b1a9-01f99a62a257

**Comment by Te-Z** [comment]
Could you update the icons colors on web ? IMO it should have the same selected/not selected colors than mobile view

image

**Comment by KhaledNjim** [comment]
> Could you update the icons colors on web ? IMO it should have the same selected/not selected colors than mobile view
> 
> image

they have different colors in figma

**Comment by KhaledNjim** [comment]
> When adding new account, the avatar in the bottom sheet not changed too
> 
>  RPReplay_Final1721794814.MP4

Fixed

[login.webm](https://github.com/user-attachments/assets/023c6352-9ab7-48aa-88ff-c1d7c293450c)

**Comment by KhaledNjim** [comment]
> i tested it again in iOS, everything's good, but the splash ink around the icon when i tap, its overlapped, it should only have 1 ink.
> 
>  RPReplay_Final1721794264.MP4

fixed

**Comment by sherlockvn** [comment]
i haved tested in iOS and review again. its good for me

**Review by sherlockvn — APPROVED** [review]

**Comment by sherlockvn** [comment]
i think you can rebase @KhaledNjim

File: lib/widgets/avatar/bottom_navigation_avatar_style.dart [hunk 8]
```diff
@@ -0,0 +1,5 @@
+class BottomNavigationAvatarStyle {
+  static const double avatarSize = 24.0;
+  static const double avatarFontSize = 10.0;
+  static const double selectedavatarBorderWidth = 2.0;
```
Comment (nqhhdev) [inline lib/widgets/avatar/bottom_navigation_avatar_style.dart:4]:
```suggestion
  static const double selectedAvatarBorderWidth = 2.0;
```

File: lib/widgets/layouts/adaptive_layout/app_adaptive_scaffold_body_view.dart [hunk 9]
```diff
@@ -24,6 +25,7 @@ class AppAdaptiveScaffoldBodyView extends StatelessWidget {
   final OnPopInvoked onPopInvoked;
   final VoidCallback onOpenSettings;
   final AbsAppAdaptiveScaffoldBodyArgs? adaptiveScaffoldBodyArgs;
+  final ValueNotifier<Profile?> currentprofile;
```
Comment (nqhhdev) [inline lib/widgets/layouts/adaptive_layout/app_adaptive_scaffold_body_view.dart:28]:
```suggestion
  final ValueNotifier<Profile?> currentProfile;
```

**Review by nqhhdev — APPROVED** [review]
LGTM

**Review by sherlockvn — APPROVED** [review]

**Comment by sherlockvn** [comment]
i tested in iOS, it worked well for me.

**Review by Te-Z — APPROVED** [review]

**Pull request merged**